onion-messaging.md

```diff
@@ -0,0 +1,143 @@
+
+Note also that there is currently no chunking requirement applied here.
+
+### CONTROL MESSAGES
+
+#### HANDSHAKE CONTROL MESSAGES
+
+The message `handshake` is sent by any peer/node not configured to act as
+directory node, to any other node it connects to, as the first message.
+The message `dn-handshake` is sent by any peer/node which is configured to
+act as a directory node, to any other node, as a response to the initial
+`handshake` message.
+(Notice that this configuration implies that directory nodes do not currently
+talk to each other).
+
+The syntax of `handshake` is:
+json serialized:
+  {"app-name": "joinmarket",
+   "directory": false,
+   "location-string": "host:port",
+   "proto-ver": 5,
+   "features": {},
+   "nick": "J5***"
+  }
+Note that `proto-ver` is the version specified as `JM_VERSION` in `jmdaemon.protocol`.
+(It has not changed for many years, it only specifies the syntax of the messages).
+The `features` field is currently empty but is provided for forwards compatibility, if directory nodes offer additional features.
+
+The syntax of `dn-handshake` is:
+json serialized:
+ {"app-name": "joinmarket",
+  "directory": true,
+  "proto-ver-min": 5,
+  "proto-ver-max": 5,
+  "features": {}
+  "accepted": true,
+  "nick": "J5**",
+  "motd": "Information about directory node"
+ }
+
+ Non-directory nodes should send `handshake` to directory nodes, upon successfully connecting, and directory nodes should return the `dn-handshake` method with `true`
+ for accepted, if and only if:
+ * the protocol version is in the accepted range
+ * the `directory` field of the peer is false
+ * the `app-name` is the string `"joinmarket"`
+ * the set of features requested is both recognized and accepted (currently: none)
+
+ In case those conditions are met, return `"accepted": true`, else return
+ `"accepted": false` and immediately disconnect the new peer.
+ (in this rejection case, the remaining fields of the `dn-handshake` message do
+ not matter, but can be kept as before for convenience).
+
+In case of a direct connection between peers (neither are directory nodes),
+the party which connects then sends the first `handshake` message, and the
+connected-to party responds with their own `handshake`.
+
+In this case, the connection should be accepted and maintained by the receiver
+if and only if:
+* the protocol version is identical
+* the `directory` field of the peer is false
+* the `app-name` is `"joinmarket"`
+* the set of features is both recognized and accepted (currently: none)
+
+otherwise the peer should be immediately disconnected.
+
+ALL OTHER MESSAGES (control or otherwise, as detailed below), cannot be sent/
+will be ignored until the above two-way handshake is complete.
+
+#### OTHER CONTROL MESSAGES
+
+The syntax of `peerlist` is:
+
+nick || NICK_PEERLOCATOR_SEPARATOR || peer-location || "," ... (repeated)
```

Comment:
This should be wrapped in ``` like the other.

Reply:
Done in https://github.com/JoinMarket-Org/JoinMarket-Docs/pull/12/commits/f36877ab359c71f972a406f3781319787dc8926a